src/AWS/Lambda/Runtime.hs

```diff
@@ -0,0 +1,62 @@
+module AWS.Lambda.Runtime
+  ( module Exports
+  , ClientError (..)
+  , LambdaResponse (..)
+  , runtime
+  )
+where
+
+import AWS.Lambda.Client
+import AWS.Lambda.Context as Exports
+import AWS.Prelude
+import Control.Exception (throwIO)
+import Control.Monad.Except (runExceptT)
+import Data.Aeson (ToJSON(..))
+
+runtime
+  :: ToJSON result
+  => (Either ClientError LambdaResponse -> IO (Either Text result))
```

Comment:
I failed to find a compelling reason to use `RIO` within the `runtime` function where the `env` would potentially be the `lambdaContext` but I am happy to reconsider since its a possible alternative but with the downside of increasing the API surface area.

Furthermore adding `lambdaContext` as an `env` value requires what I currently consider internal library implementation detail and its generation is an effect that can lead to lambda client errors.


Secondly an attempt of adding `RIO` as in code snippet below also may require I expose another function such as `runRioMruntime` to execute our new rio runtime with the appropriate env while also potentially exposing any possible generations errors to the function callback argument if we want to expose lambda client errors in one place.

```
```
mRuntime
  :: forall env result . (HasLambdaContext env, ToJSON result)
  => (Either ClientError Value -> RIO env (Either Text result))
  -> IO ()
mRuntime = ... 
```